pkg/operator/ceph/file/dependent.go

```diff
@@ -49,3 +70,43 @@ func CephFilesystemDependents(clusterdCtx *clusterd.Context, clusterInfo *client
 
 	return deps, nil
 }
+
+// return subvolume groups that have 1 or more subvolumes present in them
+func subvolumeGroupDependents(clusterdCtx *clusterd.Context, clusterInfo *client.ClusterInfo, filesystem *v1.CephFilesystem) (*dependents.DependentList, error) {
+	baseErr := "failed to get Ceph subvolume groups containing subvolumes"
+
+	deps := dependents.NewDependentList()
+
+	svgs, err := client.ListSubvolumeGroups(clusterdCtx, clusterInfo, filesystem.Name)
+	if err != nil {
+		return deps, errors.Wrap(err, baseErr)
+	}
+
+	errs := []error{}
+	for _, svg := range svgs {
+		svs, err := client.ListSubvolumesInGroup(clusterdCtx, clusterInfo, filesystem.Name, svg.Name)
+		if err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to list subvolumes in subvolume group %q", svg.Name))
+		}
+
+		if svg.Name == indexSubvolumeGroupName {
+			// TODO: there are two mutually exclusive design options here:
+			//   1. this check is future proof in the event that other metadata subvolumes are added
+			//      to '_index' like 'clone' exists today (not sure how likely this is)
+			//      (this is what is implemented currently)
+			//   2. we can explicitly only ignore the 'clone' subvolume but block if there are
+			//      like if a user deliberately put a subvolume into '_index' (unlikely)
+			logger.Debugf("skipping dependency check for subvolumes in subvolumegroup %q", indexSubvolumeGroupName)
+			continue
+		}
+
+		if len(svs) > 0 {
+			deps.Add(subvolumeGroupDependentType, svg.Name)
+		}
+	}
+
+	outErr := util.AggregateErrors(errs, "failed to list subvolumes in filesystem %q for one or more subvolume groups; "+
+		"a timeout might indicate there are many subvolumes in a subvolume group", filesystem.Name)
```

Comment:
If we're hitting a timeout due to large numbers of volumes, seems like we need a different check? Or at a minimum a longer timeout?

Reply:
I don't think so. A timeout indicates that either (a) Ceph is not responding, or (b) the number of volumes is so large that the call is taking a long time. In both cases, we still want to block deletion.